lib/DAC/DAC.cpp

```diff
@@ -0,0 +1,140 @@
+//
+// Digital to Analog Converter
+//
+
+#include "../../include/definitions.h"
+
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+#include <I2CBus.h>
+#include <Helper.h>
+#include "DAC.h"
+
+#include <Wire.h> // I2C
+#include <stdio.h>
+#include <inttypes.h>
+
+#define BASE_ADDR_CMD 0xA8
+
+
+void dac_demo_task(void *pvParameter) {
+
+    // polling loop
+    uint8_t val = 0;
+    while(1) {
+
+        // set potentiometer value
+        printf("set potentiometer value=%d", val);
+        set_pot(val, POT_CHAN0);
+        set_pot(0xFF - val, POT_CHAN1); // negate
+
+        // sleep for 100ms
+        vTaskDelay(100 / portTICK_PERIOD_MS);
+
+        // increment timer
+        val += 10;
+    }
+}
+
+void dac_user_input_demo_task(void *pvParameter) {
+
+    // polling loop
+    uint8_t val = 0;
+    int input;
+    int n = 10;
+    char input_str[n];
+
+    printf("start dac task\n");
+
+    while(1) {
+
+        // read user input
+        fgets_stdio_blocking(input_str, n);
+
+        printf("fgets read: %s\n", input_str);
+
+        sscanf(input_str, "%d", &input);
+
+        if(input >= 0 && input < 255){
```

Comment:
Why not 256?

Reply:
Good catch! 
Yes you are right, that should either be a `<=` or `256`